lib/datadog/grok/src/matchers/date.rs

```diff
@@ -219,7 +227,13 @@ pub fn time_format_to_regex(
 pub fn apply_date_filter(value: &Value, filter: &DateFilter) -> Result<Value, GrokRuntimeError> {
     match value {
         Value::Bytes(bytes) => {
-            let value = String::from_utf8_lossy(bytes);
+            let mut value = String::from_utf8_lossy(bytes).to_string();
```

Comment:
```suggestion
            let mut value = String::from_utf8_lossy(bytes).into_owned();
```

Relying on the method on `Cow<'_, str>` makes this more clear to my eye.